### ISSUE TITLE
lin_backoff is actually exponential

### ISSUE THREAD

**Issue by bamorim** [safwank/ElixirRetry#27]
Sorry to ask that question, but your `lin_backoff` is actually exponential. Actually, it yields the same numbers as  `exp_backoff` if factor is 2.

```
iex(5)> Retry.DelayStreams.exp_backoff(10) |> Enum.take(10)
[20, 40, 80, 160, 320, 640, 1280, 2560, 5120, 10240]
iex(6)> Retry.DelayStreams.lin_backoff(10, 2) |> Enum.take(10)
[20, 40, 80, 160, 320, 640, 1280, 2560, 5120, 10240]
```

I can open a PR for this, but what I'd suggest is to just have the lin_backoff named exp_backoff and set the factor default to 2.

Then we can implement a lin_backoff like that:

```
  def lin_backoff(initial_delay, factor) do
    Stream.unfold(1, fn failures ->
      next_d = initial_delay + failures * factor
      {next_d, failures + 1}
    end)
  end
```

What do you think?

EDIT: fixed typo (`last_delay` -> `initial_delay`)

**Comment by Dzol** [comment]
We've observed this too 🙂 It'd be nice to see a fix for it

**Comment by safwank** [comment]
I'll look into it this weekend, fingers crossed.

**Comment by safwank** [comment]
@bamorim @Dzol Sorry it took me awhile, but here's what I'm thinking of doing.

- deprecate `lin_backoff/2` to maintain backwards compatibility
- create a new function that actually returns linear delays
```elixir
  def linear_backoff(delay \\ 100) do
    Stream.repeatedly(fn -> delay end)
  end
```

What do you think?

**Comment by bamorim** [comment]
@safwank then this is not a linear backoff but rather a constant backoff. IMO, linear should increase by a constant rate `a`, starting with some value `b`, so then it fits a curve `a*x + b`

**Comment by safwank** [comment]
@bamorim You're right. I'll rename the above function to `const_backoff/1` and introduce a new one called `linear_backoff/2` that really really :) returns a stream of linearly increasing delays.

**Comment by safwank** [comment]
@bamorim If you want, you can review the latest commits before I release v0.11.

**Comment by safwank** [comment]
Fixed in [v0.11](https://github.com/safwank/ElixirRetry/releases/tag/v0.11.0).
